### DISCUSSION TITLE
Any way to override Application theme on a window?

### DISCUSSION THREAD

**Discussion by kuiperzone** [AvaloniaUI/Avalonia#7001] [Q&A] [answered]
Hi,

I'd like to override to override the Application theme on just one window, so if the main theme is Fluent light, the window would be Fluent dark.

I've tried this:

    <Window xmlns="https://github.com/avaloniaui"
         ...
        WindowStartupLocation="CenterScreen">

        <Window.Styles>
            <FluentTheme Mode="Dark"/>
        </Window.Styles>

        ....

    </Window>

While it shows the window in dark mode, many of the controls are invisible. Perhaps my approach is naive.

Is there any relatively simple way to achieve this?

**Comment by workgroupengineering** [comment, marked as answer]
I think the only solution is to remove the style from Application.Styles and set it for each window.

**Reply by kuiperzone** [reply]
Thanks, that might work for me.

**Comment by kekekeks** [comment]
You can override `IStyleHost::StylingParent` in your `Window` and return null. That will disable global styling for that particular window.

**Reply by kuiperzone** [reply]
Even better. Thanks!

**Reply by kuiperzone** [reply]
Just a quick follow on question...

How does one override `IStyleHost::StylingParent`? It does not appear to be a virtual method.

**Reply by kuiperzone** [reply]
Don't worry. Recon I've figured it out my last question...

    public class DarkWindow : Window, IStyleHost
    {
        .... 
        IStyleHost? IStyleHost.StylingParent { get { return null; } }
    }
